common/models/role.js

```diff
@@ -244,31 +256,32 @@ module.exports = function(Role) {
         return;
       }
       debug('Model found: %j', inst);
-      var ownerId = inst.userId || inst.owner;
-      // Ensure ownerId exists and is not a function/relation
-      if (ownerId && 'function' !== typeof ownerId) {
-        if (callback) callback(null, matches(ownerId, userId));
-        return;
```

Comment:
I am open to discuss how to handle multiple `belongsTo` relations, but for the purpose of this pull request, I prefer to stay focused on fixing the issue created by the previous patch, i.e. "allowing users from different models but with the same id), because the `Role.isOwner()` method was not checking the user's principalType."